frontend/src/components/smart-contracts/ContractInteractionProvider.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useEffect, useReducer } from 'react';
+import { useWallet } from 'components/wallet/useWallet';
+import { ERC20ApprovalProvider } from 'components/smart-contracts/ERC20ApprovalProvider';
+import {
+  smartContractReducer,
+  initialSmartContractsState,
+  SmartContractActions,
+} from 'components/smart-contracts/SmartContractReducer';
+import { SmartContractContext } from 'components/smart-contracts/SmartContractContext';
+
+export const ContractInteractionProvider = ({
```

Comment:
Imho - we should just have a `ContractsProvider` that provides everything.

Reply:
Agreed.  I renamed this and removed the reducer.